airflow/models/__init__.py

```diff
@@ -1739,7 +1669,7 @@ def email_alert(self, exception):
         jinja_context.update(dict(
             exception=exception,
             exception_html=exception_html,
-            try_number=self.try_number - 1,
+            try_number=self.try_number,
```

Comment:
Why did we change this?

Reply:
This was because the test order did change because I moved the DagBag tests to a separated file. In the testing the state of a task would be changed from `RUNNING` to something else. the `try_number` in `TaskInstances` does a +1 based on the state. Not the cleanest way. @ashb, I see this part is on your name. Can you double check the logic here?

Reply:
About all I remember of this code now was it was a hack on top of a hack :(